File: P1S3/src/CadenaFiltros.cpp
#include "CadenaFiltros.h"

CadenaFiltros::CadenaFiltros(){
	this->target = nullptr;
}

CadenaFiltros::CadenaFiltros(Post &target){
	this->target = &target;
}

void CadenaFiltros::addFiltro(Filtro &filtro) {
	filtros.push_back(&filtro);
}

void CadenaFiltros::setTarget(Post &objetivo){
    target = &objetivo;
}

void CadenaFiltros::ejecutar() {
    for(int i= 0; i < filtros.size(); i++)
            filtros[i]->ejecutar(*target);       
}

File: P2/CadenaFiltros.dart
import 'Filtro.dart';
import 'Post.dart';

class CadenaFiltros {

  List<Filtro> filtros;
  Post target;


  CadenaFiltros(Post target){
    this.target = target;
    this.filtros = [];
  }

  addFiltro(Filtro filtro) {
    filtros.add(filtro);
  }

  setTarget(Post objetivo){
    target = objetivo;
  }

  ejecutar() {
    for(int i= 0; i < filtros.length; i++)
      filtros[i].ejecutar(target);
  }
}


